machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+        "## Data Transformations\n",
+        "\n",
+        "ML.NET supports a variety of data transformations that will convert data into the required format and help you make corrections to your data. Some common operations are manipulating columns, normalizing values, replacing missing values, converting values, and more. \n",
+        "\n",
+        "Look for more information on [data transformations available](https://docs.microsoft.com/en-us/dotnet/machine-learning/resources/transforms). \n",
+        "\n",
+        "Below are a few common transformations. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### One Hot Encoding \n",
+        "\n",
+        "One hot encoding is an important step for data containing strings. ML algorithms require data to be numerical, it doesn't know how to process a string. The columns of vendor_id and payment_type are categorical, vendor can be \"CMD\" or \"VST\" and payment can be \"CReDit\" or \"CaSH\". One hot encoding takes the string values passed in and converts them into numerical data."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "var pipeline = mlContext.Transforms.Categorical.OneHotEncoding(\n",
+        "    new[] { new InputOutputColumnPair(@\"vendor_id\", @\"vendor_id\"), \n",
+        "    new InputOutputColumnPair(@\"payment_type\", @\"payment_type\")},\n",
+        "    outputKind: OutputKind.Binary); \n",
+        "                 "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Replace missing values \n",
+        "\n",
+        "Another common operation is to replace missing values. Here we use the default replacement mode, which replaces the value with the default value for its type."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "pipeline.Append(mlContext.Transforms.ReplaceMissingValues(\n",
+        "    new[] { new InputOutputColumnPair(@\"rate_code\", @\"rate_code\"), \n",
+        "    new InputOutputColumnPair(@\"passenger_count\", @\"passenger_count\"), \n",
+        "    new InputOutputColumnPair(@\"trip_time_in_secs\", @\"trip_time_in_secs\"), \n",
+        "    new InputOutputColumnPair(@\"trip_distance\", @\"trip_distance\") })); \n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Now let's concatenate all of our feature columns. "
```

Comment:
Recommend adding a small note why you would want to do this

One reason:  Many ML trainers expect features to be of vector type. Mainly because applying operations to a vector is more efficient.